src/asl.ts

```diff
@@ -0,0 +1,720 @@
+      return {
+        [this.getStateName(stmt)]: this.eval(stmt.expr),
+      };
+    } else if (stmt.kind === "ThrowStmt") {
+      return {
+        [this.getStateName(stmt)]: {
+          Type: "Pass",
+          Next: this.throw(stmt),
+        },
+      };
+    } else if (stmt.kind === "VariableStmt") {
+      if (stmt.expr === undefined) {
+        return {};
+      }
+      return {
+        [this.getStateName(stmt)]: this.eval(stmt.expr),
+      };
+    } else if (stmt.kind === "TryStmt") {
+      return {
+        ...this.execute(stmt.tryBlock),
+        ...(stmt.catchClause ? this.execute(stmt.catchClause) : {}),
+        ...(stmt.finallyBlock ? this.execute(stmt.finallyBlock) : {}),
+      };
+    } else if (stmt.kind === "CatchClause") {
+      return this.execute(stmt.block);
+    } else {
+      return assertNever(stmt);
+    }
+
+    throw new Error(`cannot convert Statement '${stmt.kind}' to ASL`);
+  }
+
+  /**
+   * Find the next state that the {@link node} should transition to.
+   */
+  private next(node: FunctionlessNode | undefined): string {
+    if (node === undefined) {
+      throw new Error(`Stack Underflow`);
+    } else if (isExpr(node)) {
+      return this.next(node.parent);
+    } else if (isStmt(node)) {
+      if (node.kind === "TryStmt" && node.finallyBlock) {
+        return this.getStateName(node.finallyBlock.statements[0]);
+      } else if (node.kind === "ReturnStmt") {
+        return this.return(node);
+      } else if (node.next) {
+        return this.getStateName(node.next);
+      }
+    } else if (node.kind === "FunctionDecl") {
+      return "Success";
+    }
+    return this.next(node.parent);
+  }
+
+  /**
+   * Find the nearest state on the call stack to return the value to.
+   */
+  private return(node: FunctionlessNode | undefined): string {
+    if (node === undefined) {
+      throw new Error(`Stack Underflow`);
+    } else if (node.kind === "FunctionDecl") {
+      return "Success";
+    } else if (node.kind === "FunctionExpr") {
+      if (!this.returnTransitions.has(node)) {
+        throw new Error(`Not Implemented`);
+      }
+      return this.returnTransitions.get(node)!;
+    } else {
+      return this.return(node.parent);
+    }
+  }
+
+  /**
+   * Find the nearest state on the call stack to throw an error to.
+   */
+  private throw(node: FunctionlessNode | undefined): string {
+    if (node === undefined) {
+      throw new Error(`Stack Underflow`);
+    } else if (node.kind === "FunctionDecl") {
+      return "Throw";
+    } else if (node.kind === "FunctionExpr") {
+      return this.getThrowTransition(node);
+    } else if (node.kind === "TryStmt") {
+      if (node.catchClause) {
+        return this.next(node.catchClause);
+      } else if (node.finallyBlock) {
+        return this.next(node.finallyBlock);
+      } else {
+        return this.throw(node.parent);
+      }
+    } else {
+      return this.throw(node.parent);
+    }
+  }
```

Comment:
This works for explicit throws, but what about ay other implicit error/

Reply:
The Catch block of every task will point at a state designated for handling error transitions.